pkg/gatherers/clusterconfig/gather_monitoring_pvs.go

```diff
@@ -0,0 +1,142 @@
+package clusterconfig
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/openshift/insights-operator/pkg/record"
+	"github.com/openshift/insights-operator/pkg/utils"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	coreV1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"sigs.k8s.io/yaml"
+)
+
+// GatherMonitoringPVs Collects Persistent Volumes from openshift-monitoring namespace
+// which matches with ConfigMap configuration yaml
+//
+// ### API Reference
+// - https://github.com/kubernetes/client-go/blob/master/kubernetes/typed/core/v1/configmap.go
+// - https://github.com/kubernetes/client-go/blob/master/kubernetes/typed/core/v1/persistentvolume.go
+//
+// ### Sample data
+// - docs/insights-archive-sample/config/persistentvolumes/pvc-{hash-id}.json
+//
+// ### Location in archive
+// - `config/persistentvolumes/{persistent_volume_name}.json`
+//
+// ### Config ID
+// `clusterconfig/monitoring_persistent_volumes`
+//
+// ### Released version
+// - 4.13
+//
+// ### Backported versions (TODO: update backported releases)
+// - 4.12
+// - 4.11
+//
+// ### Changes
+// None
+func (g *Gatherer) GatherMonitoringPVs(ctx context.Context) ([]record.Record, []error) {
+	kubeClient, err := kubernetes.NewForConfig(g.gatherProtoKubeConfig)
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	mg := MonitoringPVGatherer{ctx: ctx, client: kubeClient.CoreV1()}
+
+	name, errors := mg.getDefaultPrometheusName()
+	if len(errors) > 0 {
+		return nil, errors
+	}
+
+	fmt.Printf("name: %v\n", name)
+
+	return mg.gather(name)
+}
+
+type MonitoringPVGatherer struct {
+	ctx    context.Context
+	client coreV1.CoreV1Interface
+}
+
+// getDefaultPrometheusName returns prometheus name as it's described on the configmap
+// or an error collection from the attempts to retrieve that information
+func (mg MonitoringPVGatherer) getDefaultPrometheusName() (string, []error) {
+	const CMO = "cluster-monitoring-config"
+	const NAMESPACE = "openshift-monitoring"
+
+	cm, err := mg.client.ConfigMaps(NAMESPACE).Get(mg.ctx, CMO, metaV1.GetOptions{})
+	if err != nil {
+		return "", []error{err}
+	}
+
+	var errors []error
+	for i := range cm.Data {
```

Comment:
I am probably missing something, but why do you need to iterate over the data here please?

Reply:
Data brings a collection, so I prepared the code to have more than one configuration yaml files.
If having more than one is a bad practice, I can check length and go for the first item directly.